tests/unit/deferred.py

```diff
@@ -0,0 +1,222 @@
+# Copyright 2016 Massachusetts Open Cloud Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the
+# License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS
+# IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+'''Functional test for deferred.py'''
+
+import pytest
+import tempfile
+
+from hil import config, deferred, model
+from hil.model import db, Switch
+from hil.test_common import config_testsuite, config_merge, \
+                             fresh_database, fail_on_log_warnings
+from flask import Flask
+from flask_sqlalchemy import SQLAlchemy
+
+fail_on_log_warnings = pytest.fixture(autouse=True)(fail_on_log_warnings)
+fresh_database = pytest.fixture(fresh_database)
+
+DeferredTestSwitch = None
+
+
+class SwitchError(Exception):
+    pass
+
+
+def new_db():
+    """ returns a new database connection"""
+    local_app = Flask(__name__.split('.')[0])
+    uri = config.cfg.get('database', 'uri')
+    local_app.config.update(SQLALCHEMY_TRACK_MODIFICATIONS=False)
+    local_app.config.update(SQLALCHEMY_DATABASE_URI=uri)
+    local_db = SQLAlchemy(local_app)
+    return local_db
+
+
+@pytest.fixture
+def configure():
+    config_testsuite()
+
+    additional_config = {
+        'extensions': {
+            'hil.ext.obm.mock': ''
+            }
+        }
+
+    # if we are using sqlite's in memory db, then change uri to a db on disk
+    uri = config.cfg.get('database', 'uri')
+    if uri == 'sqlite:///:memory:':
+        with tempfile.NamedTemporaryFile() as temp_db:
+            uri = 'sqlite:///' + temp_db.name
+            additional_config['database'] = {'uri': uri}
+            config_merge(additional_config)
+            config.load_extensions()
+            yield
+    else:
+        config_merge(additional_config)
+        config.load_extensions()
+        yield
+
+
+@pytest.fixture()
+def _deferred_test_switch_class():
```

Comment:
We should add a docstring explaining why this is a fixture, instead of just declaring a class once at global scope -- that's something that's not immediately apparent from the code.